### PR TITLE
feat(storage): Add Info method returns object information from header

### PR DESCRIPTION
**Pull request by curzolapierre**
- [x] Add a changelog entry in `CHANGELOG.md`

`Info` method returns `ObjectNotFound` custom error in case of object not found

### CODE REVIEW

File: storage/backend.go [hunk 1]
```diff
@@ -22,6 +22,7 @@ type Backend interface {
 	Upload(ctx context.Context, file io.Reader, path string) error
 	Size(ctx context.Context, path string) (int64, error)
 	Delete(ctx context.Context, path string) error
+	IsObjectExists(ctx context.Context, objectName string) (bool, error)
```
Comment (Soulou) [inline storage/backend.go:25]: What about having a method `Info` which return a custom error if it does not exist?

(I'm not a fan of the `IsObjectExists` function name --'

File: storage/s3.go [hunk 2]
```diff
@@ -177,6 +179,37 @@ func (s *S3) Delete(ctx context.Context, path string) error {
...
+			if apiErr.ErrorCode() == NotFoundErrCode {
+				return types.Info{}, &ObjectNotFound{}
+			}
+		}
+		return types.Info{}, errors.Wrapf(err, "fail to HEAD object '%v'", path)
+	}
+
+	return types.Info{
+		ContentLength: res.ContentLength,
+		ContentType:   *res.ContentType,
```
Comment (Soulou) [inline storage/s3.go:209]: Normally both API would also have a MD5/SHA256, could you check that and include it?

**Review by Soulou — CHANGES_REQUESTED** [review]

File: storage/s3.go [hunk 3]
```diff
@@ -177,6 +179,38 @@ func (s *S3) Delete(ctx context.Context, path string) error {
...
+				return types.Info{}, ObjectNotFound{}
+			}
+		}
+		return types.Info{}, errors.Wrapf(err, "fail to HEAD object '%v'", path)
+	}
+
+	return types.Info{
+		ContentLength: res.ContentLength,
+		ContentType:   *res.ContentType,
+		Checksum:      *res.ChecksumSHA256,
```
Comment (Soulou) [inline storage/s3.go:210]: Please add the type of CHecksum too ;-) ChecksumSHA256, it may evolve over time.

THanks!

**Review by Soulou — CHANGES_REQUESTED** [review]

**Comment by curzolapierre** [comment]
Indeed it was a little more complicate than expected. Now it should be understood, thanks :+1: 

I put this here in case for me in the future: https://docs.aws.amazon.com/AmazonS3/latest/userguide/checking-object-integrity.html#checking-object-integrity-md5
About this ^
I guess logs-archives are not impacted by the 16MB limitation, but maybe we are in other case (e.g. backups).
If we want to check the integrity of each part of the upload we will have to support it specifically.

**Review by Soulou — APPROVED** [review]

**Pull request merged**